lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -210,14 +220,163 @@ void HWMemSimImplPass::generateMemory(HWModuleOp op, FirMemory mem) {
   outputOp->setOperands(outputs);
 }
 
+// Get all the hierarchichal names for a HWModuleOp.
+void static getHierarchichalNames(HWModuleOp op,
+                                  SmallVector<std::string> &names,
+                                  const std::string path,
+                                  mlir::SymbolUserMap &symbolUsers) {
+
+  if (!op)
+    return;
+  bool parentFound = false;
+  for (auto u : symbolUsers.getUsers(op)) {
+    if (auto inst = dyn_cast<InstanceOp>(u)) {
+      auto opPath = inst.instanceName().str() + "." + path;
+      getHierarchichalNames(inst->getParentOfType<HWModuleOp>(), names, opPath,
+                            symbolUsers);
+      parentFound = true;
+    }
+  }
+  if (!parentFound)
+    names.push_back(op.getName().str() + "." + path);
+}
```

Comment:
Does this functionality exist anywhere already? I think we have this for `getAbsolutePaths` in the FIRRTL `InstanceGraph`, but that won't work here.